### PR TITLE
ENH first pass at mdet regression test

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds a regression test to the test suite. It runs mdet on a simple sim and makes sure the catalogs come out the same.

### CODE REVIEW

File: mdet_tests/test_mdet_regression.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        )
+        fnames = glob.glob(pth)
+        assert len(fnames) > 0
+        for fname in fnames:
+            old_data = fitsio.read(fname)
+            for col in old_data.dtype.names:
+                if np.issubdtype(old_data[col].dtype, np.number):
+                    assert np.allclose(
+                        all_res[col], old_data[col],
+                        atol=2e-8, rtol=1e-5,
```
Comment (beckermr) [inline mdet_tests/test_mdet_regression.py:233]:
```suggestion
                        atol=3e-8, rtol=1e-5,
```

File: mdet_tests/test_mdet_regression.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        )
+        fnames = glob.glob(pth)
+        assert len(fnames) > 0
+        for fname in fnames:
+            old_data = fitsio.read(fname)
+            for col in old_data.dtype.names:
+                if np.issubdtype(old_data[col].dtype, np.number):
+                    assert np.allclose(
+                        all_res[col], old_data[col],
+                        atol=5e-8, rtol=1e-5,
```
Comment (beckermr) [inline mdet_tests/test_mdet_regression.py:233]:
```suggestion
                        atol=1e-6, rtol=1e-5,
```

File: mdet_tests/test_mdet_regression.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        )
+        fnames = glob.glob(pth)
+        assert len(fnames) > 0
+        for fname in fnames:
+            old_data = fitsio.read(fname)
+            for col in old_data.dtype.names:
+                if np.issubdtype(old_data[col].dtype, np.number):
+                    assert np.allclose(
+                        all_res[col], old_data[col],
+                        atol=1e-6, rtol=1e-5,
```
Comment (beckermr) [inline mdet_tests/test_mdet_regression.py:233]:
```suggestion
                        atol=1e-5, rtol=1e-5,
```

File: mdet_tests/test_mdet_regression.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        )
+        fnames = glob.glob(pth)
+        assert len(fnames) > 0
+        for fname in fnames:
+            old_data = fitsio.read(fname)
+            for col in old_data.dtype.names:
+                if np.issubdtype(old_data[col].dtype, np.number):
+                    assert np.allclose(
+                        all_res[col], old_data[col],
+                        atol=1e-5, rtol=1e-5,
```
Comment (beckermr) [inline mdet_tests/test_mdet_regression.py:233]:
```suggestion
                        atol=2e-6, rtol=1e-5,
```

**Comment by beckermr** [comment]
So the benchmark here is from running version 1.3.8 on osx.

If I run the same version on linux, I can get set a lower tolerance (around 3-e8). 

For what is on master now, a value at 2e-6 is needed for the test to pass. 

Thus we have already seen some changes, but they are at a level that is probably ok to ignore.

We could also change the test to benchmark against what is in master now as well. 

What do you think @esheldon?

**Comment by esheldon** [comment]
should we remake the reference file on linux?

**Comment by beckermr** [comment]
> should we remake the reference file on linux?

Sure. This won't change much but is probably good for reproducing things.

File: .github/workflows/mdet.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          pip install git+https://github.com/esheldon/metadetect.git \
+            --force-reinstall --no-deps --no-cache-dir
+
+          python -m pip install -e .
+
+      - name: test
+        shell: bash -l {0}
+        run: |
+          pip install git+https://github.com/esheldon/ngmix.git@master
+          cd mdet_tests
```
Comment (beckermr) [inline .github/workflows/mdet.yml:55]:
```suggestion
          pushd mdet_tests
```

**Review by beckermr — COMMENTED** [review]
One but

**Comment by beckermr** [comment]
OK. This is ready to go. The tests check against 1.3.8 and the current code on master.

**Comment by esheldon** [comment]
I think the test itself looks good.

Why is there also a notebook?

My main complaint is that there are now multiple versions of this rather large file in the history.

**Comment by beckermr** [comment]
We can squash down the history. The notebook was for me to make a quick plot of the simulated image to make sure it works.

**Comment by beckermr** [comment]
Let me change the test to build the benchmark on the fly.

**Comment by beckermr** [comment]
OK ready for another look. Do not merge with squshing.

**Comment by beckermr** [comment]
I removed the big files in the repo here. We ok to merge?

**Pull request merged**